actor-tests/src/test/scala/org/apache/pekko/pattern/AskSpec.scala

```diff
@@ -266,6 +266,29 @@ class AskSpec extends PekkoSpec("""
       val (promiseActorRefForSelection, "ask") = p.expectMsgType[(ActorRef, String)]: @unchecked
       promiseActorRefForSelection.path.name should startWith("_user_myName")
     }
+
+    "proper path when promise actor terminated" in {
+      implicit val timeout: Timeout = Timeout(300 millis)
+      val p = TestProbe()
+
+      val act = system.actorOf(Props(new Actor {
+          def receive = {
+            case _ =>
+              val senderRef: ActorRef = sender()
+              senderRef ! "complete"
+              p.ref ! senderRef
+          }
+        }), "pathPrefix")
```

Comment:
For this tests, I think we would better test it several times with some generated pathPrefixs.

Reply:
random str is ok?

Reply:
I don't think it's necessary to verify the `pathPrefixs` several times, because this is not a flaky behavior. 

After investigation, I think it is a clear bug, and the unit test explicitly verifies this. Let me explain:

The reason why the path is correct after adding STDOUT is because `toString` is called `path`

https://github.com/apache/incubator-pekko/blob/d884540c92d0d38797c32f217107d97e15cdafbe/actor-typed/src/main/scala/org/apache/pekko/actor/typed/internal/ActorRefImpl.scala#L52C5-L52C62

Which initialized the path with refPathPrefix(correctly path):

https://github.com/apache/incubator-pekko/blob/d884540c92d0d38797c32f217107d97e15cdafbe/actor/src/main/scala/org/apache/pekko/pattern/AskSupport.scala#L593-L597

But when toString is not called in the `messageFactory` function apply, all this will not happen. When the Actor terminated, the method call to the path will generate a path without refPathPrefix (that is the place of the bug, and the place where the PR is fixed)

Reply:
This test was enough for this bug. We completed the ask future(which made PromiseActor terminated) on the Actor message handler, and then sent PromiseActorRef to the probe for saving reference.

Reply:
for me, it's enough for the bugfix.